src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/HashProviderDispenser.Unix.cs

```diff
@@ -130,7 +130,11 @@ public unsafe HmacHashProvider(IntPtr algorithmEvp, byte[] key)
 
                 fixed (byte* keyPtr = key)
                 {
-                    Check(Interop.libcrypto.HMAC_Init(out _hmacCtx, keyPtr, key.Length, algorithmEvp));
+                    _hmacCtx = Interop.Crypto.HmacCreate(keyPtr, key.Length, algorithmEvp);
+                    if (_hmacCtx == null)
```

Comment:
This if/throw could be replaced with a call to Interop.libcrypto.CheckValidOpenSslHandle.